src/cryptsetup/cryptsetup-generator.c

```diff
@@ -467,6 +467,26 @@ static crypto_device *get_crypto_device(const char *uuid) {
         return d;
 }
 
+static crypto_device *validate_uuid_and_get_crypto_device(const char *uuid, const char *key, int *ret_error)
+{
```

Comment:
see coding style: { on the same line as function name please

Reply:
error handling is borked. All across our case base we return errors as function return values, and return any other objects as parameters. i.e. `ret_error` should be moved to be the function return value, and the `static crypto_device` should become a `ret_crypt_device` parameter.

Also see https://systemd.io/CODING_STYLE about error handling

Reply:
Counter proposal: What If I get rid of both helpers ('validate_uuid_and_get_crypto_device'  and 'get_uuid_candidate') completely. I think the original code was much cleaner and I quite didn't understand what you meant when you suggested me to split out helpers. Instead I've added just very simple one around id128_is_valid